src/ucp/tag/tag_send.c

```diff
@@ -250,17 +258,22 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_tag_send_nbx,
     ucs_trace_req("send_nbx buffer %p count %zu tag %"PRIx64" to %s",
                   buffer, count, tag, ucp_ep_peer_name(ep));
 
-    if (ucs_likely(!(param->op_attr_mask &
-                     (UCP_OP_ATTR_FIELD_DATATYPE | UCP_OP_ATTR_FLAG_NO_IMM_CMPL)))) {
+    attr_mask = param->op_attr_mask &
+                (UCP_OP_ATTR_FIELD_DATATYPE | UCP_OP_ATTR_FLAG_NO_IMM_CMPL);
+
+    if (attr_mask == 0) {
         status = UCS_PROFILE_CALL(ucp_tag_send_inline, ep, buffer, count, tag);
-        if (ucs_likely(status != UCS_ERR_NO_RESOURCE)) {
-            ret = UCS_STATUS_PTR(status); /* UCS_OK also goes here */
-            goto out;
-        }
+        TAG_SEND_CHECK_STATUS(status, ret, {goto out;});
```

Comment:
1. UCP_TAG_SEND_CHECK_STATUS
2. remove { } and ; , just `TAG_SEND_CHECK_STATUS(status, ret, goto out);`